src/lib/__tests__/agents.test.ts

```diff
@@ -87,10 +87,13 @@ describe('agents', () => {
     await aliceAgent.sendMessageToConnection(aliceConnections[0], message);
 
     const bobMessages = await poll(
-      () => {
-        console.log(`Getting Bob's connection messages...`);
-        const connections = bobAgent.getConnections();
-        return connections[0].messages;
+      async () => {
+        console.log(`Getting Bob's messages from Alice...`);
+        const messages = await bobAgent.basicMessageRepository.findByQuery({
```

Comment:
I guess this will be hidden behind some higher level method, like `agent.getMessageBy(...)`, right?

Reply:
uhm, it's just not the messages that the user would want access to. I might want access to the connections that were persisted for instance. Do you suggest we have getXBy for all record types?